File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/mvp/di/modules/ServicesModule.java
package com.woowrale.mvp.di.modules;

import com.woowrale.mvp.data.repository.ApiClient;
import com.woowrale.mvp.data.repository.ApiService;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class ServicesModule {

    @Provides
    @Singleton
    ApiService getApiService(){
        return new ApiClient().getClient().create(ApiService.class);
    }
}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/di/modules/ServicesModule.kt
package com.woowrale.mvp.di.modules

import com.woowrale.mvp.data.repository.ApiClient
import com.woowrale.mvp.data.repository.ApiService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ServicesModule {

    @Provides
    @Singleton
    fun getApiServices():ApiService{
        return ApiClient.getclient().create(ApiService::class.java)
    }
}
